atomkraft/chain/node.py

```diff
@@ -240,6 +233,30 @@ def _execute(
                 raise RuntimeError(f"Non-zero return code {rt}\n{err.decode()}")
             return (out, err)
 
+    def _execute_and_json(
+        self,
+        args: List[str],
+        *,
+        stdin: Optional[bytes] = None,
+    ) -> Optional[Dict]:
+        stdout, stderr = self._execute(
+            args,
+            stdin=stdin,
+            stdout=PIPE,
+            stderr=PIPE,
+        )
+        if stdout.strip() and stderr.strip():
+            raise RuntimeWarning(
+                f"Got non-empty output on stdout and stderr:\n\n{stdout.decode()}\n\n{stderr.decode()}"
+            )
+        try:
```

Comment:
For clarity and modularity of the code, I would keep the json part in another function, even if you have to make two  calls to `_execute` and then to the json handling function. The rest looks good.

Reply:
so you suggest decoupling execute and JSON handling from the same function?

Reply:
@hvanz, made some changes reflecting on your comment :slightly_smiling_face: